main.tf

```diff
@@ -1,7 +1,7 @@
 locals {
   port                 = var.port == "" ? (var.engine == "aurora-postgresql" ? 5432 : 3306) : var.port
   db_subnet_group_name = var.db_subnet_group_name == "" ? join("", aws_db_subnet_group.this.*.name) : var.db_subnet_group_name
-  stored_creds         = var.db_creds_path == "" ? {} : jsondecode(data.aws_ssm_parameter.stored_db_creds[0].value)
+  stored_creds         = var.db_creds_path == "" ? "" : jsondecode(data.aws_ssm_parameter.stored_db_creds[0].value)
```

Comment:
jsondecode could produce any of Hash, List, String, or nil, depending on what value is stored in the `stored_db_creds[0].value`.. but it looks like it's treated like it's expected to be a password on the next line. Are you confident it should always contain a json-encoded *string*?

Reply:
yeah - it's only used on the next line in the inner ternary:

```
  master_password      = (var.create_cluster && var.create_random_password && var.is_primary_cluster) ? 
(var.db_creds_path == "" ? 
   random_password.master_password[0].result : 
   local.stored_creds) : 
var.password
```

Reply:
Right, I just wasn't sure because it looks like the original author expected it to be an object they'd need to pull specific credentials *out of*, so I was curious what writes that stored value and how much you trust it to be consistent :-)

Reply:
I mean, the _only_ usage of it is in a _scalar_ fashion, and the value is expected to be the _password_. I think String type is the only reasonable thing that it _should_ be set to.